crates/bench/benches/special.rs

```diff
@@ -48,6 +51,31 @@ fn custom_module_benchmarks(c: &mut Criterion) {
     }
 }
 
+fn custom_db_benchmarks(c: &mut Criterion) {
+    let runtime = start_runtime();
+
+    let config = Config {
+        storage: Storage::Memory,
+        fsync: spacetimedb::db::FsyncPolicy::Never,
+    };
+    let module = runtime.block_on(async { BENCHMARKS_MODULE_DB.load_module(config).await });
+    let mut group = c.benchmark_group("special");
+    // This bench take long, so adjust for it
+    group.measurement_time(Duration::from_secs(60 * 2));
+    group.sample_size(10);
+
+    for n in [10, 100] {
+        let args = sats::product![n];
+        group.bench_function(&format!("db_game/load={n}"), |b| {
+            b.iter_batched(
```

Comment:
Instead of doing `iter_batched`, try using `iter_custom` here. Then, you can:
- initialize the database
- start a timer
- run the `ai_agent_loop` `n` times (`n` will be passed as an argument to your `iter_custom` closure), advancing the current timestamp passed to the benchmark as you go.
- end the timer and return

Similar to how you were doing before.
In particular, I think right now this benchmark is mainly measuring initialization time. Adding more loops of the ai agent loop will allow us to measure that instead.

Reply:
I have done as suggested.